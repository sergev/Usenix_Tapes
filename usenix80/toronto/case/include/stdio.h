#define	BUFSIZ	512
#define	_NFILE	20
# ifndef FILE
extern	struct	_iobuf {
	char	*_ptr;
	int	_cnt;
	char	*_base;
	char	_flag;
	char	_file;
} _iob[_NFILE];
# endif

#define	_IOREAD	01
#define	_IOWRT	02
#define	_IONBF	04
#define	_IOMYBUF	010
#define	_IOEOF	020
#define	_IOERR	040
#define	_IOSTRG	0100
#define	NULL	0
#define	FILE	struct _iobuf
#define	EOF	(-1)

#define	stdin	(&_iob[0])
#define	stdout	(&_iob[1])
#define	stderr	(&_iob[2])
#define	getc(p)		(--(p)->_cnt>=0? *(p)->_ptr++&0377:_filbuf(p))
#define	getchar()	getc(stdin)
#define	putc(x,p)	(--(p)->_cnt>=0? (*(p)->_ptr++=(x)):_flsbuf((x),p))
#define	putchar(x)	putc(x,stdout)
#define	feof(p)		(((p)->_flag&_IOEOF)!=0)
#define	ferror(p)	(((p)->_flag&_IOERR)!=0)
#define	fileno(p)	p->_file

FILE	*fopen();
FILE	*freopen();

#define	_U	01
#define	_L	02
#define	_A	03
#define	_N	04
#define	_S	010

extern	char	_ctype[];

#define	isalpha(c)	(_ctype[c]&_A)
#define	isupper(c)	(_ctype[c]&_U)
#define	islower(c)	(_ctype[c]&_L)
#define	isdigit(c)	(_ctype[c]&_N)
#define	isspace(c)	(_ctype[c]&_S)
#define	toupper(c)	((c)-'a'+'A')
#define	tolower(c)	((c)-'A'+'a')
