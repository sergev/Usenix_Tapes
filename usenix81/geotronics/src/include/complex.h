/*
	complex.h -- header for complex arithmetic library routines


	This source file may be made available to others, so long as
	Geotronics Corporation (Austin, Texas) is given proper credit.


				REVISION HISTORY

09-May-1981	D A Gwyn	Created.


Usage:
	See source listing for "complx.c"
*/

typedef struct
	{
	double	r ;
	double	i ;
	}	complex ;		// complex number

extern	complex	*CAdd();
extern	complex	*CConjug();
extern	complex	*CConst();
extern	complex	*CCopy();
extern	complex	*CDiv();
extern	complex	*CMul();
extern	complex	*CPhasor();
extern	complex	*CScale();
extern	complex	*CSub();
extern	double	Modulus();
extern	double	Phase();
