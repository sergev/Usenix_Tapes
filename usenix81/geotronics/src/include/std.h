/*
	std.h -- standard header for all C programs


	This source file may be made available to others, so long as
	Geotronics Corporation (Austin, Texas) is given proper credit.


				REVISION HISTORY

09-May-1981	D A Gwyn	Created.

Usage:
	#include	<std.h> 	in your C program source.

Note:
	This file is maintained by the Manager of Software Development.
*/

// Extended data types

typedef int	bool ;			// logical data type
#define 	NO	0
#define 	YES	1

// Universal constants

#define	LOG10E	0.43429448190325182765112891891660508229439700580367
#define	PI	3.14159265358979323846264338327950288419716939937511

// Useful macros

#define	Abs( a )	((a) >= 0 ? (a) : - (a))
#define	Max( a , b )	((a) >= (b) ? (a) : (b))
#define	Min( a , b )	((a) >= (b) ? (b) : (a))
