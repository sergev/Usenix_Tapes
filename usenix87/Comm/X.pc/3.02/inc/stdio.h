/**
*
* This header file defines the information used by the standard I/O
* package.
*
**/
#define _BUFSIZ 512		/* standard buffer size */
#define BUFSIZ 512		/* standard buffer size */
#define _NFILE 20		/* maximum number of files */

struct _iobuf
{
char *_ptr;			/* current buffer pointer */
int _rcnt;			/* current byte count for reading */
int _wcnt;			/* current byte count for writing */
char *_base;		 	/* base address of I/O buffer */
char _flag;			/* control flags */
char _file;			/* file number */
int _size;			/* size of buffer */
char _cbuff;			/* single char buffer */
char _pad;			/* (pad to even number of bytes) */
};

extern struct _iobuf _iob[_NFILE];

#define _IOREAD 1		/* read flag */
#define _IOWRT 2		/* write flag */
#define _IONBF 4		/* non-buffered flag */
#define _IOMYBUF 8		/* private buffer flag */
#define _IOEOF 16		/* end-of-file flag */
#define _IOERR 32		/* error flag */
#define _IOSTRG 64
#define _IORW 128		/* read-write (update) flag */

#if SPTR
#define NULL 0			/* null pointer value */
#else
#define NULL 0L
#endif
#define FILE struct _iobuf	/* shorthand */
#define EOF (-1)		/* end-of-file code */

#define stdin (&_iob[0])	/* standard input file pointer */
#define stdout (&_iob[1])	/* standard output file pointer */
#define stderr (&_iob[2])	/* standard error file pointer */

#define getc(p) (--(p)->_rcnt>=0? *(p)->_ptr++:_filbf(p))
#define getchar() getc(stdin)
#define putc(c,p) (--(p)->_wcnt>=0? ((int)(*(p)->_ptr++=(c))):_flsbf((c),p))
#define putchar(c) putc(c,stdout)
#define feof(p) (((p)->_flag&_IOEOF)!=0)
#define ferror(p) (((p)->_flag&_IOERR)!=0)
#define fileno(p) (p)->_file
#define rewind(fp) fseek(fp,0L,0)
#define fflush(fp) _flsbf(-1,fp)
#define clearerr(fp) clrerr(fp)

FILE *fopen();
FILE *freopen();
long ftell();
char *fgets();

#define abs(x) ((x)<0?-(x):(x))
#define max(a,b) ((a)>(b)?(a):(b))
#define min(a,b) ((a)<=(b)?(a):(b))

