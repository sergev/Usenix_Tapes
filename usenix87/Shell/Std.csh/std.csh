#! /bin/csh -f
unset verbose
set outfile
set unkflag
set files
foreach argument ( $* )
    while ( 1 )
        switch ( $argument )
            case "-" :
                echo 'NULL FLAG'
                break
            case "-v*":
		set verbose 
		breaksw
            case "-o*":    
                set outfile = `expr $argument : '-.\(.*\)'` || set outfile = UNKNOWN
                break
            case "-*":    
                echo Unknown flag: \"$argument\"
                set unkflag = "$unkflag $argument"
                break
            case "*":    
                switch ( UNKNOWN )
                    case "$outfile":
			set outfile = $argument
			breaksw
                    default:        
			set files = "$files $argument" 
                endsw
                break
        endsw
        set argument = -`expr "$argument" : '-.\(.*\)'` || break
    end
end

# The following is just for testing this standard script
echo ' Argument analysis to' $0 ':'
echo Flags: -v = ${?verbose}, -o = \"$outfile\"
echo Unknown flags: $unkflag
echo Files: $files
exit

