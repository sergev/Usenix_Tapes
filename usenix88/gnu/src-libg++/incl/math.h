// This may look like C code, but it is really -*- C++ -*-
/* 
Copyright (C) 1988 Free Software Foundation
    written by Doug Lea (dl@rocky.oswego.edu)

This file is part of GNU CC.

GNU CC is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY.  No author or distributor
accepts responsibility to anyone for the consequences of using it
or for whether it serves any particular purpose or works at all,
unless he says so in writing.  Refer to the GNU CC General Public
License for full details.

Everyone is granted permission to copy, modify and redistribute
GNU CC, but only under the conditions described in the
GNU CC General Public License.   A copy of this license is
supposed to have been given to you along with GNU CC so you
can know your rights and responsibilities.  It should be in a
file named COPYING.  Among other things, the copyright notice
and this notice must be preserved on all copies.  
*/

/*
 * the value HUGE should be checked against the one in /usr/include/math.h
 */

#ifndef MATHH
#define MATHH

extern double acos(double);
extern double acosh(double);
extern double asin(double);
extern double asinh(double);
extern double atan(double);
extern double atan2(double, double);
extern double atanh(double);
extern double cbrt(double);
extern double ceil(double);
extern double copysign(double,double);
extern double cos(double);
extern double cosh(double);
extern double drem(double,double);
extern double erf(double);
extern double erfc(double);
extern double exp(double);
extern double expm1(double);
extern double fabs(double);
extern double finite(double);
extern double floor(double);
extern double frexp(double, int*);
extern double gamma(double);
extern double hypot(double,double);
extern double infnan(int);
extern int    isinf(double);
extern int    isnan(double);
extern double j0(double);
extern double j1(double);
extern double jn(int, double);
extern double ldexp(double, int);
extern double lgamma(double);
extern double log(double);
extern double log10(double);
extern double log1p(double);
extern double logb(double);
extern double modf(double, int*);
extern double pow(double, double);
extern double rint(double);
extern double scalb(double, int);
extern double sin(double);
extern double sinh(double);
extern double sqrt(double);
extern double tan(double);
extern double tanh(double);
extern double y0(double);
extern double y1(double);
extern double yn(int, double);

#if defined(sun)
struct exception
{
  int type;
  char* name;
  double arg1, arg2, retval;
};
int matherr(exception*);
#include <values.h>             // sun already provides some of defs below

#endif                          // sun


#ifndef HUGE
#define HUGE    1.701411733192644270e38 // correct for vax
#endif

/* These seem to be sun & sysV names of these constants */


#ifndef M_E
#define M_E         2.7182818284590452354
#endif
#ifndef M_LOG2E
#define M_LOG2E     1.4426950408889634074
#endif
#ifndef M_LOG10E
#define M_LOG10E    0.43429448190325182765
#endif
#ifndef M_LN2
#define M_LN2       0.69314718055994530942
#endif
#ifndef M_LN10
#define M_LN10      2.30258509299404568402
#endif
#ifndef M_PI
#define M_PI        3.14159265358979323846
#endif
#ifndef M_PI_2
#define M_PI_2      1.57079632679489661923
#endif
#ifndef M_1_PI
#define M_1_PI      0.31830988618379067154
#endif
#ifndef M_PI_4
#define M_PI_4      0.78539816339744830962
#endif
#ifndef M_2_PI
#define M_2_PI      0.63661977236758134308
#endif
#ifndef M_2_SQRTPI
#define M_2_SQRTPI  1.12837916709551257390
#endif
#ifndef M_SQRT2
#define M_SQRT2     1.41421356237309504880
#endif
#ifndef M_SQRT1_2
#define M_SQRT1_2   0.70710678118654752440
#endif

#ifndef PI                      // as in stroustrup
#define PI  M_PI
#endif
#ifndef PI2
#define PI2  M_PI_2
#endif

#endif                          // MATHH
