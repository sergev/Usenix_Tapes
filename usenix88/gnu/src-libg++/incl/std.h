// This may look like C code, but it is really -*- C++ -*-
/* 
Copyright (C) 1988 Free Software Foundation
    written by Doug Lea (dl@rocky.oswego.edu)

This file is part of GNU CC.

GNU CC is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY.  No author or distributor
accepts responsibility to anyone for the consequences of using it
or for whether it serves any particular purpose or works at all,
unless he says so in writing.  Refer to the GNU CC General Public
License for full details.

Everyone is granted permission to copy, modify and redistribute
GNU CC, but only under the conditions described in the
GNU CC General Public License.   A copy of this license is
supposed to have been given to you along with GNU CC so you
can know your rights and responsibilities.  It should be in a
file named COPYING.  Among other things, the copyright notice
and this notice must be preserved on all copies.  
*/


#ifndef STDH
#define STDH


extern void     _exit(int);
extern void     abort();
extern int      abs(int);
extern int      access(const char*, int);
extern int      acct(const char*);
extern unsigned alarm(unsigned);
extern void*    alloca(int);
extern double   atof(const char*);
extern int      atoi(const char*);
extern long     atol(const char*);
extern int      bcmp(const void*, const void*, int);
extern int      bcopy(const void*, void*, int);
extern void*    brk(void*);
extern int      bzero(void*, int);
extern void*    calloc(unsigned, unsigned);
extern void     cfree(void*);
extern int      chdir(const char*);
extern int      chmod(const char*, int);
extern int      chown(const char*, int, int);
extern long     clock();
extern int      close(int);
extern int      creat(const char*, int);
extern char*    crypt(const char*, const char*);
extern char*    ctermid(char*);
extern char*    cuserid(char*);
extern double   drand48();
extern int      dup(int);
extern int      dup2(int, int);
extern int      dysize(int);
extern char*    ecvt(double, int, int*, int*);
extern char*    encrypt(char*, int);
extern char**   environ;
extern double   erand(short*);
extern int      errno;
extern int      execl(const char*, ...);
extern int      execle(const char*, ...);
extern int      execlp(const char*,  ...);
extern int      exect(const char*,  char**,  char**);
extern int      execv(const char*,  char**);
extern int      execve(const char*, const char**, const char**);
extern int      execvp(const char*,  char**);
extern void     exit(int);
extern int      fchmod(int, int);
extern int      fchown(int, int, int);
extern int      fcntl(int, int, int);
extern char*    fcvt(double, int, int*, int*);
extern int      ffs(int);
extern int      flock(int, int);
extern int      fork();
extern void     free(void*);
extern int      fsync(int);
extern long     ftok(const char*, int);
extern int      ftruncate(int, unsigned long);
extern char*    gcvt(double, int, char*);
extern char*    getcwd(char*, int);
extern int      getdomainname(char*, int);
extern int      getdtablesize();
extern int      getegid();
extern char*    getenv(const char*);
extern int      geteuid();
extern int      getgid();
extern int      getgroups(int, int*);
extern long     gethostid();
extern int      gethostname(char*, int);
extern char*    getlogin();
extern int      getopt(int, char**, const char*);
extern int      getpagesize();
extern char*    getpass(const char*);
extern int      getpgrp();
extern int      getpid();
extern int      getppid();
extern int      getpriority(int, int);
extern int      getpw(int, char*);
extern int      getuid();
extern char*    getwd(const char*);
extern char*    index(const char*, int);
extern char*    initstate(unsigned, char*, int);
extern int      ioctl(int, int, char*);
extern int      isatty(int);
extern long     jrand48(short*);
extern int      kill(int, int);
extern int      killpg(int, int);
extern void     lcong48(short*);
extern int      link(const char*, const char*);
extern int      lock(int, int, long);
extern long     lrand48();
extern long     lseek(int, long, int);
extern void*    malloc(unsigned);
extern void*    memalign(unsigned, unsigned);
extern void*    memccpy(void*, const void*, int, int);
extern void*    memchr(const void*, int, int);
extern int      memcmp(const void*, const void*, int);
extern void*    memcpy(void*, const void*, int);
extern void*    memset(void*, int, int);
extern int      mkdir(const char*, int);
extern int      mknod(const char*, int, int);
extern char*    mkstemp(char*);
extern char*    mktemp(char*);
extern long     mrand48();
extern int      nice(int);
extern long     nrand48(short*);
extern int      open(const char*, int, int);
extern char*    optarg;
extern int      opterr;
extern int      optind;
extern void     pause();
extern void     perror(const char*);
extern int      pipe (int);
extern void     profil(char*, int, int, int);
extern int      psignal(unsigned, char*);
extern int      ptrace(int, int, int, int);
extern int      putenv(const char*);
extern int      rand();
extern long     random();
extern int      read(int, void*, unsigned);
extern int      readlink(const char*, char*, int);
extern void*    realloc(void*, unsigned);
extern int      rename(const char*, const char*);
extern char*    rindex(const char*, int);
extern int      rmdir(const char*);               
extern void*    sbrk(int);              
extern short*   seed48(short*);
extern int      send(int, char*, int, int);
extern int      setgid(int);
extern int      sethostname(char*, int);
extern int      setkey(const char*);
extern int      setpgrp(int);
extern int      setpriority(int, int, int);
extern int      setregid(int, int);
extern int      setreuid(int, int);
extern char*    setstate(char*);
extern int      setuid(int);
extern int      sigblock(int);
extern int      siginterrupt(int, int);
extern int      sigpause(int);
extern int      sigsetmask(int);
extern unsigned sleep(unsigned);
extern int      srand(int);
extern void     srand48(long);
extern int      stime(long*);
extern char*    strcat(char*, const char*);
extern char*    strchr(const char*, int);
extern int      strcmp(const char*, const char*);
extern char*    strcpy(char*, const char*);
extern int      strcspn(const char*, const char*);
extern char*    strdup(const char*);
extern int      strlen(const char*);
extern char*    strncat(char*, const char*, int);
extern int      strncmp(const char*, const char*, int);
extern char*    strncpy(char*, const char*, int);
extern char*    strpbrk(const char*, const char*);
extern char*    strrchr(const char*, int);
extern int      strspn(const char*, const char*);
extern double   strtod(const char*, char**);
extern char*    strtok(char*, const char*);
extern long     strtol(const char*, char**, int);
extern void     swab(void*, void*, int);
extern int      symlink(const char*, const char*);
extern char*    sys_errlist[];
extern int      sys_nerr;                  
extern int      syscall(int, ...);
extern int      system(const char*);
extern char*    tempnam(char*, char*);
extern long     time(long*);
extern char*    timezone(int, int);
extern char*    tmpnam(char*);
extern int      truncate(const char*, unsigned long);
extern char*    ttyname(int);
extern int      ttyslot();
extern unsigned ualarm(unsigned, unsigned);
extern long     ulimit(int, long);
extern int      umask(int);
extern int      unlink(const char*);
extern unsigned usleep(unsigned);
extern int      vadvise(int);
extern void*    valloc(unsigned);
extern int      vfork();
extern int      vhangup();
extern int      wait(int*);
extern int      write(int, const void*, unsigned);


#endif                          // STDH
