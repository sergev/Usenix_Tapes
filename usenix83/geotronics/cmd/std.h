/*
	<std.h> -- standard header for Geotronics Corporation C programs


	This source file may be made available to others, so long as
	Geotronics Corporation (Austin, Texas) is given proper credit.


				REVISION HISTORY

09-May-1981	D A Gwyn	Created.
30-Nov-1981	D A Gwyn	Added DEGRAD.
09-Jan-1982	D A Gwyn	Added NULL.
12-Jan-1982	D A Gwyn	Added `character', EOS.
01-Mar-1982	D A Gwyn	Added Round().
17-Mar-1982	D A Gwyn	Added `void'.
01-May-1982	D A Gwyn	`void' removed for new compiler.


Usage:
	#include	<math.h>	(if you invoke Round())
	#include	<std.h> 	in your C program source.

Note:
	This file is maintained by the Manager of Software Development.
*/

/* Extended data types */

#ifndef NULL
#define NULL	(char *)0		/* null pointer value */
#endif

typedef int	bool;			/* boolean data type */
#define 	NO	0
#define 	YES	1

typedef short	character;		/* character data type used
					   to hold chars + EOF etc. */
#define 	EOS	'\0'		/* end-of-string marker */

/* typedef int	void;			/* non-value function type;
					   remove for UNIX System III */

/* Universal constants */

#define DEGRAD	57.2957795130823208767981548141051703324054724665642
					/* degrees per radian */
#define LOG10E	0.43429448190325182765112891891660508229439700580367
#define PI	3.14159265358979323846264338327950288419716939937511

/* Useful macros */

#define Abs( a )	((a) >= 0 ? (a) : -(a))
#define Max( a, b )	((a) >= (b) ? (a) : (b))
#define Min( a, b )	((a) >= (b) ? (b) : (a))
#define Round( dbl )	floor( (dbl) + 0.5 )
