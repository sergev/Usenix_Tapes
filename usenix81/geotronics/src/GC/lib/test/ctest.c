/*
 * ctest.c - complex arithmetic test
 *
 *	last edit: 10-Dec-1980	D A Gwyn
 */

#include	<stdio.h>
#include	<math.h>
#include	<complex.h>
#include	<std.h>


main()
	{
	complex	a , b , *cp ;

	/* CConst test */
	cp = CConst( 3.0 , 4.0 , &a );
	printf( "CConst; %f s.b. 3.0 , %f s.b. 4.0\n" , a.r , a.i );
	printf( "CConst; %f s.b. 3.0 , %f s.b. 4.0\n" , cp->r , cp->i );

	/* CCopy test */
	cp = CCopy( &a , &b );
	CConst( 1.0 , sqrt( 3.0 ) , &a );
	printf( "CCopy; %f s.b. 3.0 , %f s.b. 4.0\n" , b.r , b.i );
	printf( "CCopy; %f s.b. 3.0 , %f s.b. 4.0\n" , cp->r , cp->i );

	/* Modulus test */
	printf( "Modulus; %f s.b. 5.0\n" , Modulus( &b ) );

	/* Phase test */
	printf( "Phase; %f s.b. 60.0\n" , Phase( &a ) * DEGRAD );

	/* try other quadrants */
	a.r = - a.r ;
	printf( "Modulus; %f s.b. 2.0\n" , Modulus( &a ) );
	printf( "Phase; %f s.b. 120.0\n" , Phase( &a ) * DEGRAD );
	a.i = - a.i ;
	printf( "Modulus; %f s.b. 2.0\n" , Modulus( &a ) );
	printf( "Phase; %f s.b. -120.0\n" , Phase( &a ) * DEGRAD );
	a.r = - a.r ;
	printf( "Modulus; %f s.b. 2.0\n" , Modulus( &a ) );
	printf( "Phase; %f s.b. -60.0\n" , Phase( &a ) * DEGRAD );

	/* CPhasor test */
	cp = CPhasor( 100.0 , -20.0 / DEGRAD , &a );
	printf( "CPhasor; %f s.b. 100.0 , %f s.b. -20.0\n" ,
			Modulus( &a ) , Phase( &a ) * DEGRAD );
	printf( "CPhasor; %f s.b. 100.0 , %f s.b. -20.0\n" ,
			Modulus( cp ) , Phase( cp ) * DEGRAD );

	/* CConjug test */
	cp = CConjug( &b );
	printf( "CConjug; %f s.b. 3.0 , %f s.b. -4.0\n" , b.r , b.i );
	printf( "CConjug; %f s.b. 3.0 , %f s.b. -4.0\n" , cp->r , cp->i );

	/* CScale test */
	cp = CScale( 2.0 , &b );
	printf( "CScale; %f s.b. 6.0 , %f s.b. -8.0\n" , b.r , b.i );
	printf( "CScale; %f s.b. 6.0 , %f s.b. -8.0\n" , cp->r , cp->i );

	/* CAdd test */
	cp = CAdd( &b , CConst( -4.0 , 11.0 , &a ) );
	printf( "CAdd; %f s.b. 2.0 , %f s.b. 3.0 \n" , a.r , a.i );
	printf( "CAdd; %f s.b. 2.0 , %f s.b. 3.0 \n" , cp->r , cp->i );

	/* CSub test */
	cp = CSub( &b , CConst( 4.0 , 7.0 , &a ) );
	printf( "CSub; %f s.b. -2.0 , %f s.b. 15.0 \n" , a.r , a.i );
	printf( "CSub; %f s.b. -2.0 , %f s.b. 15.0 \n" , cp->r , cp->i );

	/* CMul test */
	cp = CMul( CConst( 1.0 , 2.0 , &a ) , CConst( -1.0 , 3.0 , &b ) );
	printf( "CMul; %f s.b. -7.0 , %f s.b. 1.0\n" , b.r , b.i );
	printf( "CMul; %f s.b. -7.0 , %f s.b. 1.0\n" , cp->r , cp->i );

	/* CDiv test */
	cp = CDiv( &a , &b );
	printf( "CDiv; %f s.b. -1.0 , %f s.b. 3.0\n" , b.r , b.i );
	printf( "CDiv; %f s.b. -1.0 , %f s.b. 3.0\n" , cp->r , cp->i );
	}
