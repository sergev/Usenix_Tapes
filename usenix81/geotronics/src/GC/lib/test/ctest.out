CConst; 3.000000 s.b. 3.0 , 4.000000 s.b. 4.0
CConst; 3.000000 s.b. 3.0 , 4.000000 s.b. 4.0
CCopy; 3.000000 s.b. 3.0 , 4.000000 s.b. 4.0
CCopy; 3.000000 s.b. 3.0 , 4.000000 s.b. 4.0
Modulus; 5.000000 s.b. 5.0
Phase; 60.000000 s.b. 60.0
Modulus; 2.000000 s.b. 2.0
Phase; 120.000000 s.b. 120.0
Modulus; 2.000000 s.b. 2.0
Phase; -120.000000 s.b. -120.0
Modulus; 2.000000 s.b. 2.0
Phase; -60.000000 s.b. -60.0
CPhasor; 100.000000 s.b. 100.0 , -20.000000 s.b. -20.0
CPhasor; 100.000000 s.b. 100.0 , -20.000000 s.b. -20.0
CConjug; 3.000000 s.b. 3.0 , -4.000000 s.b. -4.0
CConjug; 3.000000 s.b. 3.0 , -4.000000 s.b. -4.0
CScale; 6.000000 s.b. 6.0 , -8.000000 s.b. -8.0
CScale; 6.000000 s.b. 6.0 , -8.000000 s.b. -8.0
CAdd; 2.000000 s.b. 2.0 , 3.000000 s.b. 3.0 
CAdd; 2.000000 s.b. 2.0 , 3.000000 s.b. 3.0 
CSub; -2.000000 s.b. -2.0 , 15.000000 s.b. 15.0 
CSub; -2.000000 s.b. -2.0 , 15.000000 s.b. 15.0 
CMul; -7.000000 s.b. -7.0 , 1.000000 s.b. 1.0
CMul; -7.000000 s.b. -7.0 , 1.000000 s.b. 1.0
CDiv; -1.000000 s.b. -1.0 , 3.000000 s.b. 3.0
CDiv; -1.000000 s.b. -1.0 , 3.000000 s.b. 3.0
